Pinta.Core/Extensions/CairoExtensions.cs

```diff
@@ -52,7 +54,245 @@ public readonly record struct Color (
 	{
 		public Color (double r, double g, double b)
 			: this (r, g, b, 1.0)
-		{ }
+		{
+		}
+
+		/// <summary>
+		/// Returns the color value as a string in hex color format.
+		/// </summary>
+		/// <param name="addAlpha">If false, returns in format "RRGGBB" (Alpha will not affect result).<br/>
+		/// Otherwise, returns in format "RRGGBBAA".</param>
+		public string ToHex (bool addAlpha = true)
+		{
+			int r = Convert.ToInt32 (R * 255.0);
+			int g = Convert.ToInt32 (G * 255.0);
+			int b = Convert.ToInt32 (B * 255.0);
+			int a = Convert.ToInt32 (A * 255.0);
+
+			if (addAlpha)
+				return $"{r:X2}{g:X2}{b:X2}{a:X2}";
+			else
+				return $"{r:X2}{g:X2}{b:X2}";
+		}
+
+		/// <summary>
+		/// Returns a color from an RGBA hex color. Accepts the following formats:<br/>
+		/// RRGGBBAA<br/>
+		/// RRGGBB<br/>
+		/// RGB (Expands to RRGGBB)<br/>
+		/// RGBA (Expands to RRGGBBAA)<br/>
+		/// Will accept leading #.
+		/// </summary>
+		/// <param name="hex">Hex color as a string.</param>
+		/// <returns>Resulting color. If null, the method could not parse it.</returns>
+		public static Color? FromHex (string hex)
+		{
+			if (hex.StartsWith ("#"))
+				hex = hex.Remove (0, 1);
+
+			// handle shorthand hex
+			if (hex.Length == 3)
+				hex = $"{hex[0]}{hex[0]}{hex[1]}{hex[1]}{hex[2]}{hex[2]}";
+			if (hex.Length == 4)
+				hex = $"{hex[0]}{hex[0]}{hex[1]}{hex[1]}{hex[2]}{hex[2]}{hex[3]}{hex[3]}";
+
+			if (hex.Length != 6 && hex.Length != 8)
+				return null;
+			try {
+				int r = int.Parse (hex.Substring (0, 2), NumberStyles.HexNumber);
+				int g = int.Parse (hex.Substring (2, 2), NumberStyles.HexNumber);
+				int b = int.Parse (hex.Substring (4, 2), NumberStyles.HexNumber);
+				int a = 255;
+				if (hex.Length > 6)
+					a = int.Parse (hex.Substring (6, 2), NumberStyles.HexNumber);
+				return new Color (r / 255.0, g / 255.0, b / 255.0, a / 255.0);
+			} catch {
+				return null;
+			}
+		}
+
+		/// <summary>
+		/// Copied from RgbColor.ToHsv<br/>
+		/// Returns the Cairo color in HSV value.
+		/// </summary>
+		/// <returns>HSV struct.
+		/// Hue varies from 0 - 360.<br/>
+		/// Saturation and value varies from 0 - 1.
+		/// </returns>
+		public HsvColor ToHsv ()
```

Comment:
The original code in RgbColor/HsvColor came from PDN so we should carry over the same license header into this file. Or, split this out into a separate Color.cs to make it clear that the rest of the code in CairoExtensions.cs is not under that license